core/aws_ddk_core/stages/firehose_s3.py

```diff
@@ -0,0 +1,221 @@
+        buffering_size: Optional[Size] = None,
+        compression: Optional[destinations.Compression] = None,
+        data_output_prefix: Optional[str] = None,
+        data_stream: Optional[Stream] = None,
+        enable_data_stream: Optional[bool] = False,
+        encryption_key: Optional[IKey] = None,
+        error_output_prefix: Optional[str] = None,
+        logging: Optional[bool] = True,
+        log_group: Optional[ILogGroup] = None,
+        alarm_threshold: Optional[int] = 900,
+        alarm_evaluation_periods: Optional[int] = 1,
+    ) -> None:
+        """
+        DDK Firehose to S3 stage.
+
+        Parameters
+        ----------
+        scope : Construct
+            Scope within which this construct is defined
+        id : str
+            Identifier of the stage
+        environment_id : str
+            Identifier of the environment
+        delivery_stream_name: Optional[str] = None
+            Name of the Firehose Delivery Stream
+        delivery_stream: Optional[firehose.IDeliveryStream] = None
+            Existing Delivery Stream to use in this stage
+        bucket_name: Optional[str] = None
+            Name of S3 Bucket to be created as a delivery destination
+        bucket: Optional[IBucket] = None
+            Existing S3 Bucket to use as a destination for the Firehose Stream.
+            If no bucket is provided, a new one is created
+        buffering_interval: Optional[Duration] = None
+            The length of time that Firehose buffers incoming data before delivering it to the S3 bucket.
+            Minimum: Duration.seconds(60)
+            Maximum: Duration.seconds(900)
+            Default: Duration.seconds(300)
+        buffering_size: Optional[Size] = None
+            The size of the buffer that Kinesis Data Firehose uses for incoming
+            data before delivering it to the S3 bucket.
+            Minimum: Size.mebibytes(1)
+            Maximum: Size.mebibytes(128)
+            Default: Size.mebibytes(5)
+        compression: Optional[Compression] = None
+            The type of compression that Kinesis Data Firehose uses to compress
+            the data that it delivers to the Amazon S3 bucket.
+            Default: - UNCOMPRESSED
+        data_stream: Optional[Stream] = None
+            Existing Kinesis Data Stream to use in stage before Delivery Stream.
+            Setting this parameter will override any creation of Kinesis Data Streams
+            in this stage. `enable_data_stream` will have no effect.
+        enable_data_stream: Optional[bool] = False,
+            Enable Kinesis Data Stream to fron the Firehose delivery stream.
+            Default: false
+        data_output_prefix: Optional[str] = None
+            A prefix that Kinesis Data Firehose evaluates and adds to records before writing them to S3.
+            This prefix appears immediately following the bucket name.
+            Default: “YYYY/MM/DD/HH”
+        enable_data_stream: Optional[bool] = False
```

Comment:
Nitpick - to be consistent:
https://github.com/awslabs/aws-ddk/blob/main/core/aws_ddk_core/stages/sqs_lambda.py#L46
Naming it `data_stream_enabled`